packages/ag-charts-community/src/chart/mapping/prepareSeries.test.ts

```diff
@@ -144,258 +186,560 @@ describe('transform series options', () => {
     });
 
-    test('processSeriesOptions', () => {
-        const result = processSeriesOptions({}, seriesOptions);
+    describe('#processSeriesOptions', () => {
+        describe('Stacking and grouping configuration combinations', () => {
+            const seriesTypes = {
+                area: { stackable: true, groupable: false, stackedByDefault: false },
+                column: { stackable: true, groupable: true, stackedByDefault: false },
+                line: { stackable: false, groupable: false, stackedByDefault: false },
+                nightingale: { stackable: true, groupable: true, stackedByDefault: true },
+                'range-column': { stackable: false, groupable: true, stackedByDefault: false },
+            };
 
-        expect(result).toMatchInlineSnapshot(`
-            [
-              {
-                "fill": "pink",
-                "grouped": true,
-                "seriesGrouping": {
-                  "groupCount": 4,
-                  "groupIndex": 0,
-                  "stackCount": 0,
-                  "stackIndex": 0,
+            const stackingSeriesOptions: Record<keyof typeof seriesTypes, Array<any>> = {
+                area: areas,
+                column: columns,
+                line: lines,
+                nightingale: nightingales,
+                'range-column': rangeColumns,
+            };
+
+            Object.entries(seriesTypes).forEach(([seriesType, { stackable, groupable, stackedByDefault }]) => {
+                if (stackable) {
+                    addStackableSeriesType(seriesType);
+                }
+                if (groupable) {
+                    addGroupableSeriesType(seriesType);
+                }
+                if (stackedByDefault) {
+                    addStackedByDefaultSeriesType(seriesType);
+                }
+            });
+
+            beforeEach(() => {
+                console.warn = jest.fn();
+            });
+
+            it.each(Object.keys(seriesTypes))(
+                `should handle options stacked property 'true' appropriately`,
+                (seriesType) => {
+                    const sType = seriesType as keyof typeof seriesTypes;
+                    const sOptions = stackingSeriesOptions[sType];
+                    const result: typeof sOptions = processSeriesOptions(
+                        {},
+                        sOptions.map((s) => ({ ...s, stacked: true }))
+                    );
+                    const groupable = seriesTypes[sType].groupable;
+                    const stackable = seriesTypes[sType].stackable;
+                    const stacked = stackable;
+                    const grouped = !stacked && groupable;
+                    result.forEach((s) => {
+                        if (stackable) {
+                            expect(s.stacked).toBe(stacked);
+                            expect(s.grouped).toBe(grouped);
+                        } else {
+                            expect(console.warn).toBeCalled();
```

Comment:
Nit: Can we also verify that the console warning is the expected warning message?

I think this can be achieved with something like:
```
expect(console.warn).toHaveBeenCalledWith(/some regexp matching a console message/);
```